core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala

```diff
@@ -281,6 +291,18 @@ trait RawlsGroupComponent {
       }
     }
 
+    def setGroupAsRealm(groupRef: RawlsGroupRef): ReadWriteAction[Int] = {
```

Comment:
I was imagining a scenario where admins could promote groups to be realms. I'll check with PO to see if that's something that we actually want. If not, then you're right that this is a potentially misleading/dangerous function to have

Regardless, I'll organize the whole idea of realm creation a little bit better between `UserService` and `RawlsGroupComponent`